### PR TITLE
Feature/condo/sberdoma 436/protect files

### PR DESCRIPTION
**Pull request by dkoviazin**
File protection
To test how it works rename in .env file 
```js
  { "bucket":"doma.ai.public" }
```
to
```js
  { "bucket":"private" }
```

### CODE REVIEW

File: apps/condo/domains/common/utils/sberCloudFileAdapter.js [hunk 1]
```diff
@@ -1,11 +1,59 @@
 const ObsClient = require('esdk-obs-nodejs')
+const path = require('path')
+const { SERVER_URL, SBERCLOUD_OBS_CONFIG } = require('@core/config')
+const { getItem } = require('@keystonejs/server-side-graphql-client')
+const { isEmpty } = require('lodash')
+class SberCloudObsAcl {
```
Comment (pahaz) [inline apps/condo/domains/common/utils/sberCloudFileAdapter.js:6]: It’s better to have some tests for that

File: apps/condo/index.js [hunk 2]
```diff
@@ -97,6 +99,15 @@ const authStrategy = keystone.createAuthStrategy({
     },
 })
 
+class OBSFilesMiddleware {
+    prepareMiddleware ({ keystone, dev, distDir }) {
+        const app = express()
+        // TODO(zuch): Ask if this path is ok
+        app.use('/api/files/:file(*)', obsRouterHandler({ keystone }))
```
Comment (Dimitreee) [inline apps/condo/index.js:106]: this path is ok

**Review by Dimitreee — APPROVED** [review]

File: apps/condo/index.js (hunk 2 above)
Comment (dkoviazin) [reply apps/condo/index.js:106]: Ok

File: apps/condo/domains/common/utils/sberCloudFileAdapter.js (hunk 1 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/sberCloudFileAdapter.js:6]: Added tests for set meta and get meta

File: apps/condo/domains/ticket/schema/TicketFile.js [hunk 3]
```diff
@@ -37,6 +39,25 @@ const TicketFile = new GQLListSchema('TicketFile', {
...
+                    const key = `${TICKET_FILE_FOLDER_NAME}/${filename}`
+                    // OBS will lowercase all keys from meta
+                    const metaToSet = {
+                        listkey: listKey,                    
+                        id,
+                    }
+                    try {
+                        await Adapter.acl.setMeta(key, metaToSet)
+                    } catch (error) {
+                        console.error('OBS set meta for file', error)
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/TicketFile.js:56]: It's better not to hide the error. We should not save the file in such cases

File: apps/condo/domains/common/utils/sberCloudFileAdapter.js [hunk 4]
```diff
@@ -1,11 +1,63 @@
...
```
Comment (pahaz) [inline apps/condo/domains/common/utils/sberCloudFileAdapter.js:6]:
```suggestion

class SberCloudObsAcl {
```

File: apps/condo/domains/common/utils/fileAdapter.js [hunk 5]
```diff
@@ -18,10 +18,7 @@ class FileAdapter {
             case 'sbercloud':
                 Adapter = this.createSbercloudFileApapter()
                 if (!Adapter) {
-                    // no fallback for production
```
Comment (pahaz) [inline apps/condo/domains/common/utils/fileAdapter.js:21]: We can fail without config at the moment

File: apps/condo/domains/ticket/schema/TicketFile.js (hunk 3 above)
Comment (dkoviazin) [reply apps/condo/domains/ticket/schema/TicketFile.js:56]: Done!

File: apps/condo/domains/common/utils/sberCloudFileAdapter.js (hunk 4 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/sberCloudFileAdapter.js:6]: Fixed

File: apps/condo/domains/common/utils/fileAdapter.js (hunk 5 above)
Comment (dkoviazin) [reply apps/condo/domains/common/utils/fileAdapter.js:21]: Done

**Review by AntonAL — COMMENTED** [review]
Could you please write a comment about what `obsRouterHandler` does to reduce time to understand the code by others?

**Review by AntonAL — APPROVED** [review]

**Pull request merged**